File: Workwear/Models/Company.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Workwear.Models
{
    public class Company
    {
        public int Id { get; set; }
        [Required]
        [DisplayName("Полное название")]
        public string FullName { get; set; }
        [Required]
        [DisplayName("Короткое название")]
        public string ShortName { get; set; }
        [Required]
        [DisplayName("ИНН")]
        public string INN { get; set; }
        [Required]
        [DisplayName("КПП")]
        public string KPP { get; set; }
        [Required]
        [DisplayName("ОГРН")]
        public string OGRN { get; set; }
        public string? City { get; set; }
        [Required]
        [DisplayName("Юридический адрес")]
        public string LegalAddress { get; set; }
        [Required]
        [DisplayName("Фактический адрес")]
        public string ActualAddress { get; set; }
        [Required]
        [DisplayName("Почтовый индекс")]
        public string PostalCode { get; set; }
        public string PhoneNumber { get; set; }
    }
}

File: Workwear/wwwroot/js/company.js
var dataTable;

$(document).ready(function () {
    loadDataTable();
});

function loadDataTable() {
    dataTable = $('#tblData').DataTable({
        "ajax": { url: '/admin/Company/getall' },
        "columns": [
            { "data": "fullName", "width": "10%" },
            { "data": "shortName", "width": "5%" },
            { "data": "inn", "width": "5%" },
            { "data": "kpp", "width": "5%" },
            { "data": "ogrn", "width": "5%" },
            { "data": "city", "width": "10%" },
            { "data": "legalAddress", "width": "10%" },
            {
                data: 'id',
                "render": function (data) {
                    return `<div class="w-75 btn-group" role="group">
                    <a href="/admin/company/upsert?id=${data}" class="btn btn-primary mx-2"> <i class="bi bi-pencil-square"></i> Edit</a>
                    <a onClick=Delete('/admin/company/delete/${data}') class="btn btn-danger mx-2"> <i class="bi bi-trash-fill"></i> Delete</a>
                    </div >`
                },
                "width": "20%"
            }
        ]
    });
}

function Delete(url) {
    Swal.fire({
        title: 'Are you sure?',
        text: "You won't be able to revert this!",
        icon: 'warning',
        showCancelButton: true,
        confirmButtonColor: '#3085d6',
        cancelButtonColor: '#d33',
        confirmButtonText: 'Yes, delete it!'
    }).then((result) => {
        if (result.isConfirmed) {
            $.ajax({
                url: url,
                type: "DELETE",
                success: function (data) {
                    dataTable.ajax.reload();
                    toastr.success(data.message);
                }
            })
        }
    })
}